src/Summoner/Config.hs

```diff
@@ -168,6 +171,9 @@ configT = Config
         <$> Toml.text "package" .= cpPackage
         <*> Toml.text "module"  .= cpModule
 
+    warnings :: Key -> BiToml [Text]
+    warnings = dimap Just maybeToMonoid . Toml.maybeT (Toml.arrayOf Toml._Text)
```

Comment:
We already have this `BiToml` for the list of `Text`, it can be reused 🙂 You can even rename it accordingly so its name won't confused when used with warnings as well.